src/mock/ms8.rs

```diff
@@ -0,0 +1,815 @@
+        Mersenne8(modular_pow(self.0, exp, MS8))
+    }
+}
+
+impl From<u32> for Mersenne8 {
+    #[inline]
+    fn from(v: u32) -> Mersenne8 {
+        Mersenne8::new(v)
+    }
+}
+
+impl From<u64> for Mersenne8 {
+    #[inline]
+    fn from(v: u64) -> Mersenne8 {
+        Mersenne8((v % u64::from(MS8)) as u32)
+    }
+}
+
+impl From<Mersenne8> for u32 {
+    fn from(v: Mersenne8) -> u32 {
+        v.0
+    }
+}
+
+impl PartialEq<u32> for Mersenne8 {
+    fn eq(&self, other: &u32) -> bool {
+        self.0 == *other
+    }
+}
+
+impl fmt::Display for Mersenne8 {
+    #[inline]
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        fmt::Display::fmt(&self.0, f)
+    }
+}
+
+impl rand::Rand for Mersenne8 {
+    #[inline]
+    fn rand<R: rand::Rng>(rng: &mut R) -> Self {
+        Mersenne8::from(<u32 as rand::Rand>::rand(rng))
+    }
+}
+
+impl Field for Mersenne8 {
+    #[inline]
+    fn zero() -> Self {
+        Mersenne8(0)
+    }
+
+    #[inline]
+    fn one() -> Self {
+        Mersenne8(1)
+    }
+
+    #[inline]
+    fn is_zero(&self) -> bool {
+        self.0 == 0
+    }
+
+    #[inline]
+    fn square(&mut self) {
+        self.0 = (*self * *self).0;
+    }
+
+    #[inline]
+    fn double(&mut self) {
+        // MS8 fits at least twice into a single u32.
+        self.0 = (self.0 * 2) % MS8;
+    }
+
+    #[inline]
+    fn negate(&mut self) {
+        self.0 = (Self::zero() - *self).0;
+    }
+
+    #[inline]
+    fn add_assign(&mut self, other: &Self) {
+        *self += other;
+    }
+
+    #[inline]
+    fn sub_assign(&mut self, other: &Self) {
+        *self -= other;
+    }
+
+    #[inline]
+    fn mul_assign(&mut self, other: &Self) {
+        *self *= other;
+    }
+
+    #[inline]
+    fn inverse(&self) -> Option<Self> {
+        let (d, _s, t) = ext_euclid(MS8, self.0);
+
+        // MS8 is prime, so the gcd should always be 1, unless the number itself is 0.
+        if d != 1 {
+            debug_assert_eq!(d, MS8);
+            return None;
+        }
+
+        Some(Mersenne8::new(
+            ((t + i64::from(MS8)) % i64::from(MS8)) as u32,
+        ))
+    }
+
+    #[inline]
+    fn frobenius_map(&mut self, _power: usize) {
+        // Does nothing, the frobenius endomorphism is the identity function in every finite field
+        // of prime order.
+    }
+}
+
+impl PrimeField for Mersenne8 {
+    type Repr = Self;
+
+    const NUM_BITS: u32 = 32;
+    const CAPACITY: u32 = 30;
+
+    // Not actually used.
+    const S: u32 = 0;
+
+    #[inline]
+    fn from_repr(v: Self::Repr) -> Result<Self, PrimeFieldDecodingError> {
+        Ok(v)
+    }
+
+    #[inline]
+    fn into_repr(&self) -> Self::Repr {
+        *self
+    }
+
+    #[inline]
+    fn char() -> Self::Repr {
+        // Awkward. We cannot return the characteristic itself, since it's not an element of field,
+        // but equal to its order. Instead, we panic.
+        //
+        // Note: The return type of this function should probably be `usize`.
```

Comment:
You tell me. =)

Isn't there a mismatch in mathematical types here?

Reply:
I think the intention is for `Repr` to be just a "big integer" type.
It is used internally by prime field implementations, which then apply the modulo operation.
In our case, `Repr` would be just `u32`, which is big enough.

Reply:
I'd rather keep things as is for now - otherwise, a new wrapper type needs to be added, since it's not possible to implement `PrimeFieldRepr` for `u32`.